packages/edit-site/src/components/sidebar/spacing-panel.js

```diff
@@ -35,29 +44,67 @@ function useCustomUnits( { units, contextName } ) {
 	return usedUnits.length === 0 ? false : usedUnits;
 }
 
-export default function SpacingPanel( {
-	context: { name },
-	getStyle,
-	setStyle,
-} ) {
+function useCustomSides( blockName, feature ) {
+	const support = getBlockSupport( blockName, 'spacing' );
+
+	// Skip when setting is boolean as theme isn't setting arbitrary sides.
+	if ( typeof support[ feature ] === 'boolean' ) {
+		return;
+	}
+
+	return support[ feature ];
+}
+
+function filterValuesBySides( values, sides ) {
+	if ( ! sides ) {
+		// If no custom side configuration all sides are opted into by default.
+		return values;
+	}
+
+	// Only include sides opted into within filtered values.
+	const filteredValues = {};
+	sides.forEach( ( side ) => ( filteredValues[ side ] = values[ side ] ) );
+
+	return filteredValues;
+}
+
+function useThemeValues( name, feature ) {
+	return useSelect(
+		( select ) => {
+			const baseStyles = select( editSiteStore ).getSettings()
+				.__experimentalGlobalStylesBaseStyles;
+			return baseStyles?.styles?.[ name ]?.spacing?.[ feature ];
+		},
+		[ name, feature ]
+	);
+}
+
+export default function SpacingPanel( { context, getStyle, setStyle } ) {
+	const { name } = context;
+	const showPaddingControl = useHasPadding( context );
 	const units = useCustomUnits( { contextName: name } );
+
 	const paddingValues = getStyle( name, 'padding' );
-	const setPaddingValues = ( { top, right, bottom, left } ) => {
-		setStyle( name, 'padding', {
-			top: top || paddingValues?.top,
-			right: right || paddingValues?.right,
-			bottom: bottom || paddingValues?.bottom,
-			left: left || paddingValues?.left,
-		} );
+	const themePaddingValues = useThemeValues( name, 'padding' );
```

Comment:
We already  have a `useEditorFeature` hook to be used instead of `useThemeValues` which does the same thing but doesn't take care of the entire chain of theme.json

Reply:
The idea here was to be able to reset the control in the Global Styles sidebar to the actual values set by the theme. 

Using `useEditorFeature` we get restricted to just the features in `__experimentalFeatures` rather than being able to access the styles as well. Testing `useEditorFeature` in place of `useThemeValues` here, results in `undefined`  instead of theme set values, when setting padding values under the default theme.json's "styles" property for the `core/group` context.

```json
	"styles": {
		"core/group": {
			"spacing": {
				"padding": {
					"top": "50px",
					"bottom": "50px"
				}
			}
		}
	}
```

In earlier iterations that evolved into this PR, I used a ref to reset the control back to the initial values at the time the site editor was loaded. It was pointed out that it would be more in line with user expectations to reset to whatever the theme set prior to any adjustments the user made.

If there is a better approach to meet the goal of resetting to the theme values, I'm happy to make it happen.

Reply:
I understand but I think we should be cautious here to not introduce an adhoc solution, this is a problem for all properties, not just padding and right now, resetting means unsetting the property for all the other properties, so I think  we should simplify and just do that for the moment.

Reply:
From https://github.com/WordPress/gutenberg/pull/30608#discussion_r618268847

I understand that the goal of `useThemeValues( name, property )` is to get the base values (core+theme). I believe is in our best interest to evolve `getStyles` to do this instead. For example, updating it so it can do `getStyle( name, property, 'theme' )` seems a fine choice. The rationale for that is that we want to centralize access to the shape of the data, so consumers don't need to deal with shape changes, migrations, etc. ― instead, there's a centralized place for all of that (global styles provider, getStyle method, etc).

Reply:
That sort of change to the current API will be a great well-scoped PR that can land independently from these two.

Reply:
Agreed. Thanks as always for the wise counsel @youknowriad and @nosolosw 👍 

I'll update the padding and margin support PRs to simply unset values for the time being. Then next week I'll put together a PR to update `getStyle` to offer the `theme` origin.

Reply:
This has now been updated to unset padding values rather than reset to the theme values. `useThemeValues` has been removed. https://github.com/WordPress/gutenberg/pull/30607/commits/2567acfb8f5519ff4bb6d8322d03bb0203bb1ad3

Reply:
A PR is up adding a `theme` origin to `getStyle`. https://github.com/WordPress/gutenberg/pull/31267